atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -37,6 +37,192 @@ def get_config(workflow):
+                       password=config['auth'].get('password'),
+                       dockpulp_loglevel=loglevel)
+
+
+def get_odcs(workflow):
+    return get_config(workflow)['odcs']
+
+
+def get_odcs_session(workflow):
+    config = get_odcs(workflow)
+
+    from atomic_reactor.odcs_util import ODCSClient
+
+    client_kwargs = {'insecure': config.get('insecure', False)}
+
+    if config['auth'].get('openidc_dir'):
+        token_path = os.path.join(config['auth']['openidc_dir'], 'token')
+        with open(token_path, "r") as f:
+            client_kwargs['token'] = f.read().strip()
+
+    if config['auth'].get('ssl_certs_dir'):
+        cert_path = os.path.join(config['auth']['ssl_certs_dir'], 'cert')
+        if os.path.exists(cert_path):
+            client_kwargs['cert'] = cert_path
+        else:
+            raise KeyError("ODCS ssl_certs_dir doesn't exist")
+
+    return ODCSClient(config['api_url'], **client_kwargs)
+
+
+def get_smtp(workflow):
+    return get_config(workflow)['smtp']
+
+
+def get_smtp_session(workflow):
+    config = get_smtp(workflow)
+
+    import smtplib
+    return smtplib.SMTP(config['host'])
+
+
+def get_pdc(workflow):
+    return get_config(workflow)['pdc']
+
+
+def get_pdc_session(workflow):
+    config = get_pdc(workflow)
+
+    from pdc_client import PDCClient
+    return PDCClient(server=config['api_url'], ssl_verify=not config.get('insecure', False),
+                     develop=True)
+
+
+def get_arrangement_version(workflow):
+    return get_config(workflow)['arrangement_version']
+
+
+def get_artifacts_allowed_domains(workflow):
+    return get_config(workflow)['artifacts_allowed_domains']
+
+
+def get_image_labels(workflow):
+    return get_config(workflow)['image_labels']
+
+
+def get_image_equal_labels(workflow):
+    return get_config(workflow)['image_equal_labels']
+
+
+def get_openshift(workflow):
+    return get_config(workflow)['openshift']
+
+
+def get_openshift_session(workflow, namespace, conf_file=None):
+    config = get_openshift(workflow)
+
+    from osbs.api import OSBS
+    from osbs.conf import Configuration
+
+    config_kwargs = {
+        'config_file': conf_file,
+        'openshift_url': config['url'],
+        'verify_ssl': not config.get('insecure', False),
+        'namespace': namespace,
+        'conf_section': None,
+        'cli_args': None,
+        'use_auth': False,
+    }
+
+    try:
+        config_kwargs['build_json_dir'] = get_build_json_dir(workflow)
+    except KeyError:
+        pass
+
+    if config.get('auth'):
+        if config['auth'].get('krb_keytab_path'):
+            config_kwargs['kerberos_keytab'] = config['auth'].get('krb_keytab_path')
+        if config['auth'].get('krb_principal'):
+            config_kwargs['kerberos_principal'] = config['auth'].get('krb_principal')
+        if config['auth'].get('krb_cache_path'):
+            config_kwargs['kerberos_ccache'] = config['auth'].get('krb_cache_path')
```

Comment:
Minor, but could factor this:
```
auth = config.get('auth')
if auth:
    for key in ('krb_keytab_path', 'krb_principal', 'krb_cache_path'):
        config_kwargs[key] = auth.get(key)  # or do we want: try: ... = auth[key]  except: pass

    ...
```

Reply:
can't do that as keys are different, 
what I could do is 

for key1, key2 zip(('krb_keytab_path', ....), ('kerberos_keytab', .....):
    config_kwargs[key2] = auth.get(key2)

do you want this?